src/app/cli/src/init/test_ledger_application.ml

```diff
@@ -218,7 +221,36 @@ let test ~privkey_path ~ledger_path ?prev_block_path ~first_partition_slots
   let stop_tracing =
     if tracing then (fun x -> Mina_tracing.stop () ; x) else ident
   in
+  let results = ref [] in
   let init_root = Ledger.merkle_root init_ledger in
+  let save_preparation_times time =
+    if Option.is_some benchmark then results := !results @ [ time ]
+  in
+  let save_and_dump_benchmarks final_time =
+    let calculate_mean results =
+      let preparation_steps = List.drop_last_exn results in
```

Comment:
I do not think we should be dropping anything here.

How I understand the code, final call will not go through `save_preparation_times`, hence no need to drop anything.

Reply:
Absolutely!